### PR TITLE
test(api): install opentrons and simulate

### PR DESCRIPTION
**Pull request by y3rsh**
# Overview

Here is a basic framework and some tests.
Let me know what you think.

### CODE REVIEW

File: .github/workflows/api-test-lint-deploy.yaml [hunk 1]
```diff
@@ -128,6 +126,43 @@ jobs:
...
+      - uses: 'actions/checkout@v4'
+      - name: 'Fix actions/checkout odd handling of tags'
+        if: startsWith(github.ref, 'refs/tags')
+        run: |
+          git fetch -f origin ${{ github.ref }}:${{ github.ref }}
+          git checkout ${{ github.ref }}
+      - uses: 'actions/setup-python@v4'
+        with:
+          python-version: '3.10'
+      - name: Install opentrons and shared-data
```
Comment (SyntaxColoring) [inline .github/workflows/api-test-lint-deploy.yaml]:
```suggestion
      - name: Set up package-testing
```

File: .github/workflows/api-test-lint-deploy.yaml [hunk 2]
```diff
@@ -128,6 +126,43 @@ jobs:
...
+        working-directory: package-testing
+        run: make setup
+      - name: Run the tests
+        id: test
+        working-directory: package-testing
+        run: make test
+        continue-on-error: true
```
Comment (SyntaxColoring) [inline .github/workflows/api-test-lint-deploy.yaml]: Will this `continue-on-error: true` unintentionally remove the red X in the GitHub web UI if this test fails?

https://stackoverflow.com/questions/62045967/is-there-a-way-to-continue-on-error-while-still-getting-correct-feedback

**Review by SyntaxColoring — APPROVED** [review]
Nice! Thank you!

File: .github/workflows/api-test-lint-deploy.yaml [hunk 3]
```diff
@@ -128,6 +126,43 @@ jobs:
           files: ./api/coverage.xml
           flags: api
 
+  test-package:
+    name: 'installed package tests on ${{ matrix.os }}, python ${{ matrix.python }}'
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        os: ['ubuntu-22.04', 'macos-latest']
```
Comment (sfoster1) [inline .github/workflows/api-test-lint-deploy.yaml]: this does unfortunately really need to include windows

File: package-testing/Makefile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+clean:
+	@echo "Removing the results directory $(RESULTS_DIR)"
+	rm -rf $(RESULTS_DIR) || true
+
+.PHONY: teardown
+teardown: clean
+	rm -rf $(VENV_DIR) || true
+
+.PHONY: test
+test:
```
Comment (sfoster1) [inline package-testing/Makefile]: would expect clean to be a recipe dependency up here as it is for `teardown`

File: package-testing/Makefile [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	@echo "This target expects you already have run 'make setup'"
+	$(MAKE) clean
+	./test_help.sh $(VENV_DIR) $(RESULTS_DIR)
+	./Flex_v2_19_expect_success.sh $(VENV_DIR) $(RESULTS_DIR)
+	./OT2_v2_20_expect_success.sh $(VENV_DIR) $(RESULTS_DIR)
+	./Flex_v2_16_expect_error.sh $(VENV_DIR) $(RESULTS_DIR)
+	./OT2_v6PD_expect_error.sh $(VENV_DIR) $(RESULTS_DIR)
+
+.PHONY: test1
+test1:
```
Comment (sfoster1) [inline package-testing/Makefile]: probably wants `clean` as a dep

File: package-testing/Makefile [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	@echo "Easy way to run a single test"
+	@echo "Copy a line from the test target above and run it here"
+	@echo "⚠️ Don't forget to change it back before committing ⚠️"
+	./OT2_v6PD_expect_error.sh $(VENV_DIR) $(RESULTS_DIR)
```
Comment (sfoster1) [inline package-testing/Makefile]: if you make this a variable (`./$(TEST).sh $(VENV_DIR) $(RESULTS_DIR)`) then you could run it like `make test1 TEST=OT2_v6PD_expect_error` without having to edit the file

**Review by sfoster1 — CHANGES_REQUESTED** [review]
We really need to test this on windows, perhaps most of all, unfortunately. Might need to have a powershell equivalent. Sorry to say it!

File: .github/workflows/api-test-lint-deploy.yaml (hunk 3 above)
Comment (y3rsh) [reply .github/workflows/api-test-lint-deploy.yaml]: Agreed, adding.

File: .github/workflows/api-test-lint-deploy.yaml (hunk 2 above)
Comment (y3rsh) [reply .github/workflows/api-test-lint-deploy.yaml]: I always forget this. Thank You.

**Review by sfoster1 — APPROVED** [review]
Looks awesome, thank yoU!

**Pull request merged**